codesign/archive.go

```diff
@@ -0,0 +1,52 @@
+package codesign
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/xcarchive"
+)
+
+// Archive ...
+type Archive struct {
+	archive xcarchive.IosArchive
+}
+
+// NewArchive ...
+func NewArchive(archive xcarchive.IosArchive) Archive {
+	return Archive{
+		archive: archive,
+	}
+}
+
+// IsSigningManagedAutomatically ...
+func (a Archive) IsSigningManagedAutomatically() (bool, error) {
+	return a.archive.IsXcodeManaged(), nil
+}
+
+// Platform ...
+func (a Archive) Platform() (autocodesign.Platform, error) {
+	platformName := a.archive.Application.InfoPlist["DTPlatformName"]
+	switch platformName {
```

Comment:
You could extract the logic from ReadCodesignParameters, so it can be reused here.

Reply:
Added `IosArchive.Platform` in https://github.com/bitrise-io/go-xcode/pull/141/commits/7eb170b841166e5182990b48779f8f51008a2db3